packages/runtime/container-runtime/src/test/garbageCollection.spec.ts

```diff
@@ -104,266 +68,583 @@ describe("Garbage Collection Tests", () => {
         );
     };
 
-    beforeEach(async () => {
-        mockLogger = new MockLogger();
+    describe("Inactive events", () => {
+        const inactiveObjectRevivedEvent = "GarbageCollector:inactiveObjectRevived";
+        const inactiveObjectChangedEvent = "GarbageCollector:inactiveObjectChanged";
+
+        // Waits for > deleteTimeoutMs. To be called to make sure that any unreferenced nodes have been deleted.
+        async function waitForDeleteTimeout(): Promise<void> {
+            await new Promise<void>((resolve) => {
+                setTimeout(resolve, deleteTimeoutMs + 100);
```

Comment:
It would be great to mock timeouts and make sure test code does not need to actually sit idle.
I think we have an example of that somewhere in the code. Search for sinon.useFakeTimers()